reanalysis/query_panelapp.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python3
+
+
+"""
+PanelApp Parser for Reanalysis
+
+Takes a panel ID
+For the latest content, pulls Symbol, ENSG, and MOI
+    (MOI is simplified from PanelApp enum)
+
+Optionally user can provide a date in the past
+Identify the highest panel version prior to that date
+Pull all details from the earlier version
+Store all discrepancies between earlier and current
+
+Write all output to a JSON dictionary
+"""
+
+
+import logging
+from datetime import datetime
+from typing import Any, Dict, List, Optional, Union
+import json
+
+import requests
+
+import click
+
+
+# panelapp URL constants
+PANELAPP_ROOT = 'https://panelapp.agha.umccr.org/api/v1'
+PANEL_ROOT = f'{PANELAPP_ROOT}/panels'
+PANEL_CONTENT = f'{PANEL_ROOT}/{{panel_id}}'
+ACTIVITIES = f'{PANEL_CONTENT}/activities'
+
+
+def get_json_response(url: str) -> Union[List[Dict[str, str]], Dict[str, Any]]:
+    """
+    takes a request URL, checks for healthy response, returns the JSON
+    :param url:
+    :return:
+    """
+
+    response = requests.get(url, headers={'Accept': 'application/json'})
+    response.raise_for_status()
+    return response.json()
+
+
+def get_previous_version(panel_id: str, since: datetime) -> str:
```

Comment:
This name might be a bit misleading, as this really returns the first version *after* the given time, right?

I agree with your comment that it might be better to explicitly use a panel version here, to avoid confusion.

Reply:
Yeah, this was a tricky one, as 1 minute past midnight would be 'after' a datetime with day-level granularity. Taking a start date and adding a 24hr window would also fail if there were no updates on that date. I think scrapping all of this and going with strict versioning is the best route

Reply:
removed this activities search, forcing a simpler "provide a version number, or don't do a comparison"